docs/guides/application_credentials.md

```diff
@@ -0,0 +1,44 @@
+# Managing Application Credentials
+
+Following good security practice, you must never store credentials in your application code.
+
+This guide provides some best practice to help you manage credentials safely and securely.
+
+## Accessing credentials from your application
+
+Credentials should be stored in cloud secret stores such as AWS Secrets Manager or Azure Key Vault.
```

Comment:
Given the prominence of CDP, should we add a link to the CDP secrets management document?

https://portal.cdp-int.defra.cloud/documentation/how-to/secrets.md

CDP uses AWS Secrets Manager but it's abstracted through the CDP Portal from consuming teams.